com.unity.multiplayer.mlapi/Tests/Editor/ProfilerTests.cs

```diff
@@ -0,0 +1,206 @@
+            }
+        }
+
+        private static void AssertNetworkAndTransportDataExists(PerformanceTickData profilerData)
+        {
+            Assert.IsTrue(profilerData.HasData(TestProfiler.ProfilerConstants.NetworkTestData));
+            Assert.IsTrue(profilerData.HasData(TestTransport.ProfilerConstants.TransportTestData));
+        }
+
+        private static void AssertNetworkDataExists(PerformanceTickData profilerData)
+        {
+            Assert.IsTrue(profilerData.HasData(TestProfiler.ProfilerConstants.NetworkTestData));
+            Assert.IsFalse(profilerData.HasData(TestTransport.ProfilerConstants.TransportTestData));
+        }
+
+        private static void RaiseExceptionNoTickDataEvent()
+        {
+            throw new NoTickDataException();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            ProfilerNotifier.OnPerformanceDataEvent -= AssertNetworkAndTransportDataExists;
+            ProfilerNotifier.OnPerformanceDataEvent -= AssertNetworkDataExists;
+            ProfilerNotifier.OnNoTickDataEvent -= RaiseExceptionNoTickDataEvent;
+        }
+
+        [Test]
+        public void TestSentNoData()
+        {
+            const bool useNullTransport = true;
+            TestProfiler testProfiler = SetupTestProfiler(useNullTransport);
+
+            TestProfiler.ProfilerBeginTick();
+            TestProfiler.NotifyProfilerListeners();
+
+            Assert.False(testProfiler.HasSentAnyData);
+        }
+
+        [Test]
+        public void TestNormalRegisterAndNotifyFlowNull()
+        {
+            const bool useNullTransport = true;
+            TestProfiler testProfiler = SetupTestProfiler(useNullTransport);
+
+            RegisterStaticAsserts(useNullTransport);
+
+            TestProfiler.ProfilerBeginTick();
+            testProfiler.Send();
+            TestProfiler.NotifyProfilerListeners();
+
+            Assert.IsTrue(testProfiler.HasSentAnyData);
+        }
+
+        [Test]
+        public void TestNormalRegisterAndNotifyFlow()
+        {
+            const bool useNullTransport = false;
+            TestProfiler testProfiler = SetupTestProfiler(useNullTransport);
+
+            RegisterStaticAsserts(useNullTransport);
+
+            TestProfiler.ProfilerBeginTick();
+            testProfiler.Send();
+            TestProfiler.NotifyProfilerListeners();
+
+            Assert.IsTrue(testProfiler.HasSentAnyData);
+        }
+
+        [Test]
+        public void TestDroppedRegisterAndNotifyFlow()
+        {
+            const bool useNullTransport = false;
+            TestProfiler testProfiler = SetupTestProfiler(useNullTransport);
+
+            RegisterStaticAsserts(useNullTransport);
+
+            TestProfiler.ProfilerBeginTick();
+            testProfiler.Send();
+            TestProfiler.NotifyProfilerListeners();
+
+            // Capturing data after notifying listeners is bad
+            Assert.Catch<NoTickDataException>(() =>
```

Comment:
Why are we calling Assert.Catch<NoTickDataException> twice here?

Reply:
No reason. Old coder brain ensuring it worked with unnecessary redundancy. It only needs to be once.